docs/concepts/task-queues.md

```diff
@@ -4,25 +4,70 @@ title: Task Queues
+In this case, you can learn about how to implement Task Queues within the context of the language you are writing your application in:
 
-There are multiple advantages of using a Task Queue to deliver Tasks instead of invoking an Activity Worker through a synchronous RPC:
+- [Go](/docs/go/task-queues)
+- [Java](/docs/java/task-queues)
+- [PHP](/docs/php/task-queues)
 
-- Worker doesn't need to have any open ports, which is more secure.
-- Worker doesn't need to advertise itself through DNS or any other network discovery mechanism.
-- When all Workers are down, messages are persisted in a Task Queue waiting for the Workers to recover.
+From the perspective of the system's design and how everything works under the hood, things get a little bit more complex.
+
+We intend to explain this in more detail in future system design and architecture docs.
+
+## Why Task Queues?
+
+Temporal Task Queues are a little bit different from commonly used queuing technologies.
+The main difference being that they do not require explicit registration and are created on demand.
+Task Queues are very lightweight and there is no limit to the total number of Task Queues that the system can handle.
+
+There are multiple advantages of using a Task Queue to deliver Tasks to a Worker, instead of invoking actions via a synchronous RPC.
+
+- Workers do not need to have any open ports, which is more secure.
+- Workers do not need to advertise themselves through DNS or any other network discovery mechanism.
+- When all Workers are down, messages simply persist in a Task Queue, waiting for the Workers to recover.
 - A Worker polls for a message only when it has spare capacity, so it never gets overloaded.
-- Automatic load balancing across a large number of Workers.
-- Task Queues support server side throttling. This allows you to limit the Task dispatch rate to the pool of Workers and still supports adding a Task with a higher rate when spikes happen.
-- Task Queues can be used to route a request to specific pools of Workers or even a specific process.
+- Task Queues enable a sort of automatic load balancing across a large number of Workers.
+- Task Queues support server side throttling, which enables you to limit the Task dispatching rate to the pool of Workers while still supporting Task dispatching at higher rates when spikes happen.
+- Task Queues enable what we call "Task routing", which is the routing of specific Tasks to specific Workers or even a specific process.
+
+:::note
+
+All Workers listening to a given Task Queue must have identical registration of Activities/Workflows.
+The one exception to this is during a Server upgrade, where it is okay to have registration temporarily misaligned while the binary rolls out.
+
+:::
+
+## Task routing
+
+File processing tends to be one of the best examples of why you might need to route a Task to a specific process.
```

Comment:
The Go example, at least, uses sessions to accomplish this. Java and PHP do indeed appear to do some task queue naming stuff to make it work, but at minimum having Go in here is slightly confusing b/c it manages it a different (more abstracted) way.

We probably at least want to call that out, or possibly use a different example

Reply:
Ya, good catch!